pytorch_lightning/core/lightning.py

```diff
@@ -182,7 +182,11 @@ def forward(self, x):
 
         """
         if self.trainer.is_global_zero:
```

Comment:
maybe add an argument here `prog_bar=True/False` just like we have for `self.log`? WDYT?

Reply:
I suppose that these are different cases.

For self.log prog_bar=True/False works as follows:

prog_bar=True
```self.log('my_metric', 0.427, prog_bar=True)```
Epoch 36:   9%|███▉                                      | 67/716 [01:43<16:46,  1.55s/it, loss=0.462, v_num=2, my_metric=0.427]

prog_bar=False
```self.log('my_metric', 0.427, prog_bar=False)```
Epoch 36:   9%|███▉                                      | 67/716 [01:43<16:46,  1.55s/it, loss=0.462, v_num=2]

For self.print we want not break progress bar. If we have ```self.print``` call in training step:
```
def training_step(self, batch, batch_idx):
    ...
    self.print('Hello world!')
    ...
```
It will give the following log:
...
Epoch 36:   9%|███▉                                      | 67/716 [01:43<16:46,  1.55s/it, loss=0.462, v_num=2]
Hello world!
Epoch 36:   9%|███▉                                      | 68/716 [01:45<16:44,  1.54s/it, loss=0.463, v_num=2]
Hello world!
Epoch 36:   9%|███▉                                      | 69/716 [01:46<16:43,  1.55s/it, loss=0.461, v_num=2]
...

But we want the following:
...
Hello world!
Hello world!
Epoch 36:   9%|███▉                                      | 689/716 [01:46<16:43,  1.55s/it, loss=0.461, v_num=2]


May be we can give user the choice of whether to break progress bar or not, but prog_bar=True will be confusing because it has different meaning than for self.log.

Reply:
yep, it might be, maybe some other name? `use_prog_bar` or something.. Just a suggestion. Either way is fine.

Reply:
ok I thought it writes in the progress bar as a description or something. looks good :+1: